src/util/metrics/threads_linux.rs

```diff
@@ -161,48 +154,31 @@ impl Collector for ThreadsCollector {
 
 /// Gets thread ids of the given process id.
 pub fn get_thread_ids(pid: pid_t) -> Result<Vec<pid_t>> {
-    let mut tids = Vec::new();
-    let dirs = fs::read_dir(format!("/proc/{}/task", pid))?;
-    for task in dirs {
-        let file_name = match task {
-            Ok(t) => t.file_name(),
-            Err(e) => {
-                error!("fail to read task of {}, error {:?}", pid, e);
-                continue;
-            }
-        };
-
-        let tid = match file_name.to_str() {
-            Some(tid) => match tid.parse() {
-                Ok(tid) => tid,
+    Ok(fs::read_dir(format!("/proc/{}/task", pid))?
+        .filter_map(|task| {
+            let file_name = match task {
+                Ok(t) => t.file_name(),
                 Err(e) => {
-                    error!("fail to read task of {}, error {:?}", pid, e);
-                    continue;
+                    error!("read task failed"; "pid" => pid, "error" => ?e);
```

Comment:
```suggestion
                    error!("read task failed"; "pid" => pid, "err" => ?e);
```

Reply:
`error` is a keyword in log level, it's better to avoid using it in log body.

Reply:
Please address this comment or change it back and then send another PR to refine the log format.